store/src/java/com/zimbra/cs/mailbox/MailboxManager.java

```diff
@@ -713,7 +643,7 @@ public int getMailboxCount() {
     public int[] getMailboxIds() {
         int i = 0;
         synchronized (this) {
-            Collection<Integer> col = mailboxIds.values();
+            Collection<Integer> col = cacheManager.getMailboxIds();
```

Comment:
If you update the `CacheManager` interface so that `#getMailboxIds()` returns an `List<Integer>` than you can replace this whole method with.

```java
    public int[] getMailboxIds() {
        synchronized (this) {
            return cacheManager.getMailboxIds().toArray();
        }
    }
```